internal/server/iito.go

```diff
@@ -15,3 +19,84 @@ func (s *SOMASServer) runIITOEndOfTurn() error {
 	// TODO:- IITO team
 	return nil
 }
+
+func (s *SOMASServer) runGiftSession() error {
+	s.logf("start runGiftSession")
+	giftRequestDict, err := s.getGiftRequests()
+	if err != nil {
+		return err
+	}
+	giftOffersDict, err := s.getGiftOffers(giftRequestDict)
+	if err != nil {
+		return err
+	}
+	giftHistoryDict, err := s.getGiftAcceptance(giftOffersDict)
+	if err != nil {
+		return err
+	}
+	err = s.distributeGiftHistory(giftHistoryDict)
+	if err != nil {
+		return err
+	}
+	// Process actions
+	for key, value := range giftHistoryDict {
+		s.logf("Gifts from %s: %v\n", key, value)
+	}
+	defer s.logf("finish runGiftSession")
+	return nil
```

Comment:
The defer should be at the beginning, after the `s.logf("start runGiftSession")`